components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/notifier/ExternallyDeployedApiNotifier.java

```diff
@@ -88,25 +92,23 @@ private void undeployApiWhenRetiring(APIEvent apiEvent) throws NotifierException
         try {
             APIProvider apiProvider = APIManagerFactory.getInstance().getAPIProvider(CarbonContext.
                     getThreadLocalCarbonContext().getUsername());
-            API api = apiProvider.getAPIbyUUID(apiId, apiMgtDAO.getOrganizationByAPIUUID(apiId));
+            String organization = apiMgtDAO.getOrganizationByAPIUUID(apiId);
```

Comment:
we can get from apiEvent itself

Reply:
fixed with https://github.com/wso2/carbon-apimgt/pull/12944/commits/b7cf789dd478338be6ec6a7d7587148be3551ee2